packages/vitest/src/runtime/workers/forks.ts

```diff
@@ -0,0 +1,25 @@
+import v8 from 'node:v8'
+import type { WorkerGlobalState } from '../../types/worker'
+import { createForksRpcOptions, unwrapForksConfig } from './utils'
+import { runBaseTests } from './base'
+import type { VitestWorker } from './types'
+
+class ForksBaseWorker implements VitestWorker {
+  getRpcOptions() {
+    return createForksRpcOptions(v8)
+  }
+
+  async runTests(state: WorkerGlobalState) {
+    const exit = process.exit
+    state.ctx.config = unwrapForksConfig(state.ctx.config)
+
+    try {
+      await runBaseTests(state)
+    }
+    finally {
+      process.exit = exit
```

Comment:
Another option for this is to set following in `pools/rpc.ts`:

```ts
async onWorkerExit(error, code) {
  await ctx.cancelCurrentRun('test-failure')
  await ctx.runningPromise

  await ctx.logger.printError(error, { type: 'Unexpected Exit', fullStack: true })

  process.exitCode = code || 1
  await ctx.exit(true)
},
```

Right now this can hide `process.exit()` calls that user's code executes after tests have finised, like that unclosed Vite server in vite-node's tests. 

It would also be nice to print out the test file that caused this exit.

Reply:
We should do this in a separate PR before releasing the next version. I am not sure of the impact of such a change yet.

Reply:
#4903